journalpump/journalpump.py

```diff
@@ -542,6 +550,42 @@ def filter_fields(self, data):
         return {name: val for name, val in data.items() if (name.lstrip("_").lower() in self.fields) is self.whitelist}
 
 
+class UnitLogLevel:
+    def __init__(self, name: str, config: List[Dict[str, str]]):
+        self.name = name
+        self.levels = config
+        self.known_units: Dict[str, Optional[str]] = {}
+
+    def filter_by_level(self, data):
+        entry = data
+        if isinstance(data, bytes):
+            entry = json.loads(data.decode("utf-8"))
+        if not entry:
+            return {}
+        unit = entry.get("SYSTEMD_UNIT")
+        priority = entry.get("PRIORITY")
+        if (not priority) or (not unit) or unit in self.known_units and self.known_units[unit] is None:
+            # If entry does not contain unit or priority or already has been confirmed to not have a level config
+            # we return it as is
+            return data
+
+        if unit not in self.known_units:
+            self._unit_match_level_glob(unit)
+
+        if self.known_units[unit] is not None and priority <= LOG_SEVERITY_MAPPING[self.known_units[unit]]:
+            return data
+
+        return {}
+
+    def _unit_match_level_glob(self, unit: str):
+        self.known_units[unit] = None
+        for level in self.levels:
+            if fnmatch.fnmatch(unit, level.get("service_glob", "")):
+                self.known_units[unit] = level.get("log_level")
```

Comment:
```suggestion
                self.known_units[unit] = level["log_level"]
```

I think defaulting to None leads to undesirable results.